### PR TITLE
Add Interface to exempt Drawables of logging keyboard input

### LINKED ISSUE #2121: Passwords are being logged to runtime.log
I just noticed that every key press is logged in this form...

`[runtime:debug] 27/01/2019 19:12:07: KeyDownEvent(A, False) handled by OsuPasswordTextBox.`
or
`[runtime:debug] 27/01/2019 19:17:38: KeyDownEvent(A, False) handled by SearchTextBox.`

This means that passwords could appear in the logs that we are requesting for more detail for issues opened here. I believe we shouldn't log input of hidden character textboxes!

*I'm not sure if that is a framework level issue or something that should be disabled for specific elements. Please move this to o-f if it has to be fixed there.*

### PR DESCRIPTION
**Pull request by Aergwyn**
Closes #2121

I think I've found all occasions for keyboard input. :thinking:

### CODE REVIEW

File: osu.Framework/Input/Bindings/KeyBindingContainer.cs [hunk 1]
```diff
@@ -198,7 +198,7 @@ protected virtual bool PropagatePressed(IEnumerable<Drawable> drawables, T press
                     handled = drawables.OfType<IKeyBindingHandler<T>>().FirstOrDefault(d => d.OnPressed(pressed));
             }
 
-            if (handled != null)
+            if (handled != null && !(handled is IDontLogKeyPresses))
```
Comment (peppy) [inline osu.Framework/Input/Bindings/KeyBindingContainer.cs]: not sure we need this for `KeyBindingContainer`s (and may actually want events still logged for debug purposes)

@smoogipoo thoughts?

File: osu.Framework/Input/IDontLogKeyPresses.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace osu.Framework.Input
+{
+    /// <summary>
+    /// Marker interface which allows to suppress logging of keyboard input.
+    /// </summary>
+    public interface IDontLogKeyPresses
```
Comment (UselessToucan) [inline osu.Framework/Input/IDontLogKeyPresses.cs]: I would prefer a different name. For example, `ISupressKeyPressLogging`

Comment (peppy) [reply osu.Framework/Input/IDontLogKeyPresses.cs]: Let's go with `ISuppressKeyEventLogging`

File: osu.Framework/Input/IDontLogKeyPresses.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace osu.Framework.Input
+{
+    /// <summary>
+    /// Marker interface which allows to suppress logging of keyboard input.
```
Comment (peppy) [inline osu.Framework/Input/IDontLogKeyPresses.cs]:
```suggestion
    /// Marker interface which suppresses logging of keyboard input events.
    /// Useful for password fields, where user input should not be logged.
```

**Pull request merged**